mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/scheduler/AgentsErrorMonitorActor.java

```diff
@@ -32,233 +32,37 @@
 
 package io.mantisrx.master.scheduler;
 
-import static java.util.Optional.empty;
-import static java.util.Optional.of;
-
-import akka.actor.AbstractActorWithTimers;
-import akka.actor.ActorRef;
+import akka.actor.AbstractActor;
 import akka.actor.Props;
-import com.netflix.spectator.impl.Preconditions;
 import io.mantisrx.master.events.LifecycleEventsProto;
 import io.mantisrx.master.jobcluster.job.worker.WorkerState;
-import io.mantisrx.server.master.scheduler.MantisScheduler;
-import java.time.Instant;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-import java.util.concurrent.TimeUnit;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import rx.functions.Action1;
-
-public class AgentsErrorMonitorActor extends AbstractActorWithTimers implements IAgentsErrorMonitor {
-
-    private final Logger logger = LoggerFactory.getLogger(AgentsErrorMonitorActor.class);
-    private static final long ERROR_CHECK_WINDOW_MILLIS=120000; // 2 mins
-    private static final int ERROR_CHECK_WINDOW_COUNT=3;
-    private static final long TOO_OLD_MILLIS = 3600000;
-    private static final long DISABLE_DURATION_MILLIS = 60*1000; // 1mins
-    private  Action1<String> slaveEnabler = s -> logger.warn("SlaveEnabler not initialized yet!");
-    private  Action1<String> slaveDisabler = s -> logger.warn("SlaveDisabler not initialized yet!");
-    private long too_old_mills;
-    private int error_check_window_count;
-    private long error_check_window_millis;
-    private long disableDurationMillis;
-    private final Map<String, HostErrors> hostErrorMap = new HashMap<>();
-    private static final String CHECK_HOST_TIMER_KEY = "CHECK_HOST";
-    private Optional<MantisScheduler> mantisSchedulerOptional = empty();
-    // Behavior after being initialized
-    Receive initializedBehavior;
+import lombok.extern.slf4j.Slf4j;
 
-    public static Props props(long too_old_millis, int error_check_window_count,  long error_check_window_millis, long disableDurationMillis) {
-        return Props.create(AgentsErrorMonitorActor.class, too_old_millis, error_check_window_count,error_check_window_millis, disableDurationMillis);
-    }
+/**
+ * This actor is only serving the functionality right now to report worker error.
+ */
```

Comment:
Why do we need this? The JobActor already handles these events - logs them, replaces workers, etc...

Reply:
+1